pkg/repository/local_git.go

```diff
@@ -24,6 +29,23 @@ type LocalExec struct{}
 
 func (*LocalExec) RunCommand(ctx context.Context, dir string, name string, arg ...string) ([]byte, error) {
 	c := exec.CommandContext(ctx, name, arg...)
+	c.Cancel = func() error {
```

Comment:
Could we somehow add a unit/integration test testing that this part of the code gets the proper execution?

Maybe a `RunCommand` of sleeping infinite and cancelling the context 1 ms after it's executing?